xarray/core/dataarray.py

```diff
@@ -840,6 +840,99 @@ def swap_dims(self, dims_dict):
         ds = self._to_temp_dataset().swap_dims(dims_dict)
         return self._from_temp_dataset(ds)
 
+    def expand_dims(self, dim=None, axis=None):
+        """Return a new object with an additional axis (or axes) inserted at the
+        corresponding position in the array shape.
+
+        If dim is already a scalar coordinate, it will be promoted to a 1D
+        coordinate consisting of a single value.
+
+        Parameters
+        ----------
+        dim : str or sequence of str, or None.
+            Dimensions to include on the new variable.
+            dimensions are inserted with length 1.
+            If None is passed, dimension labels are automatically generated,
+            e.g. dim0, dim1, ...
+        axis : integer, list (or tuple) of integers, or None
+            Axis position(s) where new axis is to be inserted (position(s) on
+            the result array). If a list (or tuple) of integers is passed,
+            multiple axes are inserted. In this case, dim arguments should be
+            same length list or None. If None is passed, all the axes will be
+            inserted to the start of the result array.
+
+        Returns
+        -------
+        expanded : same type as caller
+            This object, but with an additional dimension(s).
+
+        Raises
+        -------
+        ValueError:
```

Comment:
It's good form to document when functions can raise, but all of these conditions are already implied by the documentation for the `axis` argument, so I don't think you need to include this section. Essentially every Python function can raise `ValueError` when passed invalid arguments.